lsp/src/types.ml

```diff
@@ -4878,85 +4878,19 @@ module DocumentFilter = struct
 end
 
 module DocumentSelector = struct
-  type t = [ `String of string | `DocumentFilter of DocumentFilter.t ] list
-  [@@deriving_inline yojson]
+  type t = DocumentFilter.t list [@@deriving_inline yojson]
 
   let _ = fun (_ : t) -> ()
 
   let t_of_yojson =
     (let _tp_loc = "lsp/src/types.ml.DocumentSelector.t" in
```

Comment:
this let binding seems no longer needed

Reply:
It's not, but this part of the file is generated. So we would need to modify the code generator to make this cleaner.